java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java

```diff
@@ -2001,7 +2001,13 @@ public void changedUpdate(DocumentEvent e) {}
                 try {
                     doc.addDocumentListener(l);
                     l.checkCancel();
-                    errors = errorProviders.stream().flatMap(errorPorvider -> errorPorvider.computeErrors(context).stream()).collect(Collectors.toList());
+                    errors = errorProviders.stream().flatMap(provider -> {
+                        List<? extends org.netbeans.api.lsp.Diagnostic> errorsOrNull = provider.computeErrors(context);
+                        if (errorsOrNull == null) {
+                            errorsOrNull = Collections.emptyList();
+                        }
+                        return errorsOrNull.stream();
+                    }).collect(Collectors.toList());
```

Comment:
Personally I'd encode the desired behaviour in the stream operation directly but what you have will work too

```suggestion
                    errors = errorProviders.stream()
                            .map(provider -> provider.computeErrors(context))
                            .filter(Object::nonNull)
                            .flatMap(List::stream)
                            .collect(Collectors.toList());
```